### PR TITLE
[FIX] forbid to confirm a PO if SO, sourced by it, is not confirmed

### PR DESCRIPTION
**Pull request by yvaucher**
This ensure we won't create twice a move going from incoming transit location to stock

### CODE REVIEW

File: sale_quotation_sourcing_stock_route_transit/model/purchase.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        return bool(unconfirmed_sales_count)
+
+    @api.multi
+    def wkf_confirm_order(self):
+        """ Forbid confirmation of purchase order when there
+        are unconfirmed sale orders sourced by its purchase lines
+        """
+        if self.has_unconfirmed_sales():
+            raise exceptions.Warning(
+                _("You cannot confirm a purchase order. You have to confirm"
```
Comment (gurneyalex) [inline sale_quotation_sourcing_stock_route_transit/model/purchase.py]: I'd rephrase this as

"You cannot confirm the purchase order before confirming the sale orders that is sources. Currently, the following sale order(s) still need confirmation: %s"

This means that has_unconfirmed_sales need rewriting to return a list of SO names.

Comment (pedrobaeza) [inline sale_quotation_sourcing_stock_route_transit/model/purchase.py]: s/sale orders that is/sales orders that are/
s/sale order(s)/sales order(s)/

And better rename field to `unconfirmed_sales` and it can be a many2many, because this can be easily tested also, and you have the complete model for anything you need.

Comment (yvaucher) [inline sale_quotation_sourcing_stock_route_transit/model/purchase.py]: @gurneyalex you mean s/the sale orders that is sources./the sale orders _it_ sources. ?
@pedrobaeza you mean s/the sale orders that are sources./the sale orders that are source_d_. ?

It should mean the sales order is sourced by a purchase order

Comment (pedrobaeza) [inline sale_quotation_sourcing_stock_route_transit/model/purchase.py]: Well, my concerns were only about plural of words, not the meaning, sorry. Unfortunately, I'm not following these set of modules (although very interesting), because I'm overloaded, and I cannot apport too much, but when I see the notification mail, that choked me.

Comment (yvaucher) [inline sale_quotation_sourcing_stock_route_transit/model/purchase.py]: Ok then I'll go for @gurneyalex proposal taking care of plural form of sale

Comment (gurneyalex) [inline sale_quotation_sourcing_stock_route_transit/model/purchase.py]: @yvaucher yes, "that it sources".

Comment (gurneyalex) [inline sale_quotation_sourcing_stock_route_transit/model/purchase.py]: @pedrobaeza sorry for choking you with a typo

**Comment by yvaucher** [comment]
Rephrased and translation added.

**Comment by yvaucher** [comment]
that's it for the "that it" it made it too like that http://en.wikipedia.org/wiki/That_that_is_is_that_that_is_not_is_not_is_that_it_it_is

**Comment by lepistone** [comment]
:+1: thanks

**Comment by gurneyalex** [comment]
:+1:

**Comment by ghost** [comment]
Why don't you put this check in module sale_quotation_sourcing instead of sale_quotation_sourcing_stock_route_transit?

**Comment by gurneyalex** [comment]
@dufresnedavid because when you are not using transit, it works, and it did not sound like a requirement for the basic module. Upon reflection, you have a point though.

@jgrandguillaume @yvaucher: what do you think about this?

**Comment by lepistone** [comment]
@yvaucher there is still an open point about moving this to `sale_quotation_sourcing`. What do you think? I've got no opinion on this, I'm happy to merge in both cases.

**Comment by yvaucher** [comment]
Afterwards, the check might be too restrictive, before changing the PR I was waiting for feedback. I'll put this in WIP

File: sale_quotation_sourcing_stock_route_transit/i18n/sale_quotation_sourcing_stock_route_transit.pot [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline sale_quotation_sourcing_stock_route_transit/i18n/sale_quotation_sourcing_stock_route_transit.pot:1]: Remove POT file.

File: sale_quotation_sourcing_stock_route_transit/model/purchase.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+#
+#    Author: Yannick Vaucher
+#    Copyright 2015 Camptocamp SA
```
Comment (pedrobaeza) [inline sale_quotation_sourcing_stock_route_transit/model/purchase.py:4]: Short header

**Comment by pedrobaeza** [comment]
Is this going to be continued?

**Comment by yvaucher** [comment]
I'm closing it, I won't work on this soon.
